### PR TITLE
feature: Add a flag to adapt SARIF ouput issue severity for GitHub code scanning compatibility [CY-2844]

### PR DESCRIPTION
**Pull request by afsmeira**
…de scanning compatibility [CY-2844]

[CY-2844]: https://codacy.atlassian.net/browse/CY-2844

### CODE REVIEW

**Review by jllopes — DISMISSED** [review]

File: cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala [hunk 1]
```diff
@@ -39,16 +39,19 @@ object Formatter {
...
-      logger.warn(s"Could not find formatter for name $formatterName. Using ${defaultFormatter.name} as fallback.")
-      defaultFormatter
-    }
+            printStream: Option[PrintStream] = Option.empty,
+            ghCodeScanningCompat: Boolean = false): Formatter = {
 
     val stream = outputFile.map(asPrintStream).orElse(printStream).getOrElse(defaultPrintStream)
 
-    builder(stream, executionDirectory)
+    formatterName match {
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala:47]: I think if we start making special cases in the outside of the formatters it gets really bad.
What do you think about passing configurations to them and they decide if they use them?

**Review by rtfpessoa — CHANGES_REQUESTED** [review]

File: cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala (hunk 1 above)
Comment (afsmeira) [reply cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala]: Yeah, you're right. I've changed things a bit to address this.

File: cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala [hunk 2]
```diff
@@ -36,22 +38,25 @@ object Formatter {
...
-
-    builder(stream, executionDirectory)
+    formatterBuilder(formatterConfiguration)
   }
 
   private def asPrintStream(file: File) = {
     new PrintStream(new FileOutputStream(file.toJava, false))
   }
+
+  case class Configuration(printStream: PrintStream, executionDirectory: File, ghCodeScanningCompat: Boolean)
```
Comment (afsmeira) [inline cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala]: Now that I'm looking at the diff I'm not sure if there is any advantage to this wrapping.

File: cli/src/main/scala/com/codacy/analysis/cli/formatter/Json.scala [hunk 3]
```diff
@@ -14,10 +14,12 @@ import scala.util.Properties
 
 object Json extends FormatterCompanion {
   override val name: String = "json"
-  override def apply(outputStream: PrintStream, executionDirectory: File): Formatter = new Json(outputStream)
+  override def apply(configuration: Configuration): Formatter = new Json(configuration)
 }
 
-private[formatter] class Json(val stream: PrintStream) extends Formatter {
+private[formatter] class Json(configuration: Configuration) extends Formatter {
```
Comment (afsmeira) [inline cli/src/main/scala/com/codacy/analysis/cli/formatter/Json.scala]: This doesn't make a lot of sense 🤔 . I'll revert.

File: cli/src/main/scala/com/codacy/analysis/cli/formatter/Text.scala [hunk 4]
```diff
@@ -3,18 +3,19 @@ package com.codacy.analysis.cli.formatter
...
 
 object Text extends FormatterCompanion {
   override val name: String = "text"
 
-  override def apply(outputStream: PrintStream, executionDirectory: File): Formatter = new Text(outputStream)
+  override def apply(configuration: Formatter.Configuration): Formatter = new Text(configuration)
 }
 
-private[formatter] class Text(val stream: PrintStream) extends Formatter {
+private[formatter] class Text(configuration: Formatter.Configuration) extends Formatter {
```
Comment (afsmeira) [inline cli/src/main/scala/com/codacy/analysis/cli/formatter/Text.scala]: I 'll revert.

**Review by rtfpessoa — DISMISSED** [review]

**Review by rtfpessoa — APPROVED** [review]

**Pull request merged**